astronomer/providers/amazon/aws/hooks/sagemaker.py

```diff
@@ -46,3 +60,105 @@ async def describe_training_job_async(self, job_name: str) -> Dict[str, Union[st
             response: Dict[str, Any] = await client.describe_training_job(TrainingJobName=job_name)
             return response
+
+    async def describe_training_job_with_log(
+        self,
+        job_name: str,
+        positions: Dict[str, Any],
+        stream_names: List[str],
+        instance_count: int,
+        state: int,
+        last_description: Dict[str, Any],
+        last_describe_job_call: float,
+    ) -> Tuple[int, Dict[str, Any], float]:
+        """
+        Return the training job info associated with job_name and print CloudWatch logs
+
+        :param job_name: name of the job to check status
+        :param positions: A list of pairs of (timestamp, skip) which represents the last record
+            read from each stream.
+        :param stream_names: A list of the log stream names. The position of the stream in this list is
+            the stream number.
+        :param instance_count: Count of the instance created for the job initially
+        :param state: log state
+        :param last_description: Latest description of the training job
+        :param last_describe_job_call: previous job called time
+        """
+        log_group = "/aws/sagemaker/TrainingJobs"
+
+        if len(stream_names) < instance_count:
+            streams = await self.logs_hook_async.describe_log_streams_async(
+                log_group=log_group,
+                stream_prefix=job_name + "/",
+                order_by="LogStreamName",
+                count=instance_count,
+            )
+            stream_names = [s["logStreamName"] for s in streams["logStreams"]]
+            positions.update([(s, Position(timestamp=0, skip=0)) for s in stream_names if s not in positions])
+
+        if len(stream_names) > 0:
+            async for idx, event in self.get_multi_stream(log_group, stream_names, positions):
+                self.log.info(event["message"])
+                ts, count = positions[stream_names[idx]]
+                if event["timestamp"] == ts:
+                    positions[stream_names[idx]] = Position(timestamp=ts, skip=count + 1)  # pragma: no cover
+                else:
+                    positions[stream_names[idx]] = Position(timestamp=event["timestamp"], skip=1)
+
+        if state == LogState.COMPLETE:
+            return state, last_description, last_describe_job_call
+
+        if state == LogState.JOB_COMPLETE:
+            state = LogState.COMPLETE
+        elif time.time() - last_describe_job_call >= 30:
+            description = await self.describe_training_job_async(job_name)
+            last_describe_job_call = time.time()
+
+            if await sync_to_async(secondary_training_status_changed)(description, last_description):
+                self.log.info(
+                    await sync_to_async(secondary_training_status_message)(description, last_description)
+                )  # pragma: no cover
+                last_description = description  # pragma: no cover
```

Comment:
why `# pragma: no cover`

Reply:
It a static function call whatever the description we send it is logging, so for log, it was complaining about the code coverage, so I ignored this in the code coverage